stdlib/fetch.py

```diff
@@ -0,0 +1,181 @@
+                'url': 'https://example.com/world.tar.gz',
+            }, {
+                'file': './hello_world'
+            },
+        ]
+
+    With the above example, ``fetch`` will do those calls, respectively: ::
+
+        fetch_url(
+            url='https://example.com/hello.tar.gz',
+            sha256='2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824',
+        )
+
+        fetch_url(url='https://example.com/world.tar.gz')
+
+        fetch_file(file='./hello_world')
+
+    """
+    build = stdlib.build.current_build()
+
+    inputs = build.args.get('fetch', [])
+
+    for input in inputs:
+        if 'url' in input:
+            fetch_url(**input)
+        elif 'file' in input:
+            fetch_file(**input)
+        else:
+            print("Bouuuh")
+
+
+def fetch_file(file: str):
+    """Copy a file or directory tothe current build cache.
+
+    :note: The path must be relative to the build manifest. If an aboslute path is given, a ``RuntimError`` is raised.
+
+    :param file: The file or directory to copy.
+    """
+    if os.path.isabs(file):
+        raise RuntimeError("fetch_file() received an absolute path as parameter, while it expects a relative one")
+
+    build = stdlib.build.current_build()
+
+    srcpath = os.path.join(
+        os.path.dirname(build.manifest.path),
+        file,
+    )
+
+    if os.path.isdir(srcpath):
+        dstpath = os.path.join(
+            build.build_cache,
+            os.path.basename(file),
+        )
+
+        shutil.copytree(
+            srcpath,
+            dstpath,
+        )
+    else:
+        shutil.copy(
+            srcpath,
+            build.build_cache,
+        )
+
+
+def fetch_url(url: str, sha256: str = None):
+    """Download a file from an URL and ensure its integrity
+
+    The downloaded file is put in the build cache of the current build, but a copy
+    is also stored in the download cache. If :py:func:`.fetch_url` is called again
+    with the same ``url`` argument, the already-downloaded file will be copied
+    instead, avoiding any extra download.
+
+    :note: Only HTTP, HTTPS and FTP protocols are supported.
+
+    :param url: The URL pointing to the file to download.
+    :param sha256: The SHA256 used to ensure the integrity of the file.
+    """
+    build = stdlib.build.current_build()
+
+    url_object = urlparse(url)
+    filename = os.path.basename(url_object.path)
+    build_path = os.path.join(
+        build.build_cache,
+        filename,
+    )
+    install_path = os.path.join(
+        build.download_cache,
+        filename,
+    )
+
+    if not sha256:
+        stdlib.log.wlog(f"No sha256 to ensure the integrity of {url}")
+
+    if os.path.exists(install_path) and _check_sha256(install_path, sha256):
+        stdlib.log.slog(f"Cache hit for {url}")
+        shutil.copy2(
+            install_path,
+            build_path
+        )
+    else:
+        stdlib.log.ilog(f"Cache miss for {url}, fetching now...")
```

Comment:
```suggestion
        stdlib.log.ilog(f"Cache missing for {url}, fetching now...")
```

Reply:
I'll keep the previous one, it's a common expression